odc/algo/_numexpr.py

```diff
@@ -31,16 +31,27 @@ def apply_numexpr_np(
     else:
         data.update(params)
 
-    if out is None and dtype is not None:
-        # This needs to be np.ndarray
-        arrays = [x for x in data.values() if isinstance(x, np.ndarray)]
-        if len(arrays) == 0:
-            raise ValueError("Could not find any arrays on input")
+    out = ne.evaluate(expr, local_dict=data, casting=casting, order=order)
```

Comment:
With this change, any `out` parameter passed into `apply_numexpr_np()` is always ignored and overwritten.

We should probably remove `out` as a function parameter (given we aren't going to use it).

Reply:
nvm. already done. PEBKAC